harness/determined/profiler.py

```diff
@@ -229,11 +232,12 @@ def record_timing(self, metric_name: str) -> Iterator[None]:
             yield
             return
 
-        timing = Timing(metric_name, self.current_batch_idx)
-        timing.start()
+        start_time = time.perf_counter()
         yield
-        timing.end()
-        self.timings_batcher_queue.put(timing)
+        dur = time.perf_counter() - start_time
+        self.timings_batcher_queue.put(
+            NamedMeasurement(metric_name, datetime.datetime.utcnow(), self.current_batch_idx, dur)
```

Comment:
This puts the timestamp as some time after the timing has completed. I think the associated timestamp for an event should be the start time of the event.

Reply:
Yeah, noticed that. Not sure if perf counters converts well to a datetime, maybe we use time.time after all.